libs/pilight/core/common.c

```diff
@@ -111,7 +112,7 @@ void array_free(char ***array, int len) {
 	}
 }
 
-unsigned int explode(char *str, const char *delimiter, char ***output) {
+unsigned int explode(const char *str, const char *delimiter, char ***output) {
```

Comment:
Yes but this is not what ```const``` mean in function declarations. In function declarations ```const``` is a promise of that function that it will not modify the item declared as const (in this case the callers buffer of chars that str is pointing to). The compiler checks that promise and complains if explode tryies something like *str = 'x'.

The promise of the function not to modify items in a callers buffer has no impact of what the caller itself does with its buffer and the chars in it.